sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/javaCode.scala

```diff
@@ -149,6 +152,34 @@ trait Block extends JavaCode {
 
   // Concatenates this block with other block.
   def + (other: Block): Block
+
+  /**
+   * Apply a map function to each java expression codes present in this java code, and return a new
+   * java code based on the mapped java expression codes.
+   */
+  def transformExprValues(f: PartialFunction[ExprValue, ExprValue]): this.type = {
```

Comment:
am I wrong or we are not updating the `exprValues` here?

Reply:
`transformExprValues` will create a new instance `Block` because a block is immutable. So once there are change, new block should take new `exprValues`.

Reply:
Ok. Will add them in next commit. Thanks.

Reply:
Added test for it.